cmake/install_helpers.cmake

```diff
@@ -60,7 +59,7 @@ function(ginkgo_install)
     write_basic_package_version_file(
         "${Ginkgo_BINARY_DIR}/GinkgoConfigVersion.cmake"
         VERSION "${PROJECT_VERSION}"
-        COMPATIBILITY AnyNewerVersion
+        COMPATIBILITY SameMajorVersion
```

Comment:
how does it detect interface breaking?

Reply:
any find_package call using a specific version currently only checks that the installed version is at least as new as the desired version:
```
# installed Ginkgo 2.0
find_package(Ginkgo 1.3) # works
```
Since we do semantic versioning, a change in the major version points to an interface break, so in my view, we should fail already at this point if the user expects 1.3, but gets 2.0

Reply:
I see. Thanks for explanation